.github/workflows/ci-tk.yaml

```diff
@@ -22,7 +22,90 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event.number || github.sha }}
   cancel-in-progress: true
 
+env:
+  LLVM_SHA_FILE: llvm-sha.txt
+  LLVM_CACHE_NUMBER: 1  # Increase to reset cache
+  WATER_REF: 1a84aa14ee74d5015ceb41bed438de4389f59b83
+
 jobs:
+  build_llvm_linux:
+    name: Builds Linux LLVM/MLIR
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [linux-mi300-1gpu-ossci-iree-org]
+    runs-on: ${{matrix.os}}
+    defaults:
+      run:
+        shell: bash -el {0}
+    timeout-minutes: 240 # Building LLVM can take multiple hours on public GH runners
+    steps:
+      - name: Checkout Water
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: ftynse/water
+          ref: ${{ env.WATER_REF }}
+          path: water
+          fetch-depth: 0
+
+      - name: Setup Cache Vars
+        run: |
+          echo 'LLVM_SHA<<EOF' >> $GITHUB_ENV
+          cat ${GITHUB_WORKSPACE}/water/$LLVM_SHA_FILE >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+
+      - name: Setup env
+        run: |
+          sudo apt update
+          sudo apt install -y ninja-build cmake clang lld
+
+      - name: Cache LLVM-MLIR
+        id: cache-llvm-mlir
+        uses: actions/cache@v3
```

Comment:
Pin actions to commit hashes. Refs can be changed out from under us and we want to pin all dependencies for security reasons.